syserol/regression.py

```diff
@@ -36,52 +36,43 @@ def function_elastic_net(function="ADCC"):
     df_variables = df_merged.drop(["subject"] + functions, axis=1)
 
     # perform regression
-    Y_pred = elasticNetFunc(df_variables, Y)
-
+    enet = elasticNetFunc(df_variables, Y)
+    Y_pred = cross_val_predict(enet, df_variables, Y, cv=len(Y), n_jobs=-1)
     return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
 
 
 def function_prediction(tensorFac, function="ADCC", evaluation="all"):
     """ Predict functions using our decomposition and regression methods"""
     func, _ = importFunction()
+    indices = AlterIndices()
 
     Y = func[function]
     X = tensorFac[1][0]  # subjects x components matrix
-    dropped = np.nonzero(np.isnan(Y.to_numpy()))
-    X = X[np.isfinite(Y), :]
+    notAlter = np.delete(np.arange(181), np.unique(np.concatenate((indices, np.where(np.isnan(Y))[0]))))
+    dropped = np.unique(np.concatenate((np.nonzero(np.isnan(Y.to_numpy()))[0], notAlter)))
+    Y_notAlter = Y[dropped][np.isfinite(Y)]
+    X_notAlter = X[Y_notAlter.index]
+    Y_Alter = Y[indices][np.isfinite(Y)]
+    X_Alter = X[Y_Alter.index]
     Y = Y[np.isfinite(Y)]
+    X = X[Y.index]
 
-    Y_pred = elasticNetFunc(X, Y)
+    enet = elasticNetFunc(X_Alter, Y_Alter)
 
     if evaluation == "all":
+        Y_pred_Alter = enet.predict(X_Alter)
+        Y_pred_notAlter = enet.predict(X_notAlter)
+        for i, drop_idx in enumerate(notAlter):
+            Y_pred_Alter = np.insert(Y_pred_Alter, drop_idx, Y_pred_notAlter[i])
+        Y_pred = Y_pred_Alter
         return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
     elif evaluation == "Alter":
-        return accuracy_alterSubj(Y, Y_pred, dropped[0])
+        Y_pred = cross_val_predict(enet, X_Alter, Y_Alter, cv=len(Y_Alter), n_jobs=-1)
+        return Y_Alter, Y_pred, np.sqrt(r2_score(Y_Alter, Y_pred))
     elif evaluation == "notAlter":
-        return accuracy_alterSubj(Y, Y_pred, dropped[0], union=False)
+        Y_pred = enet.predict(X_notAlter)
```

Comment:
Where are you fitting the model before you predict?

Reply:
In line 60 the model is fit using ElasticNetFunc() and stored in enet, I believe

Reply:
Yes, we fit is elasticNetFunc